src/r2b2/brla.py

```diff
@@ -1,151 +1,176 @@
+        """Compute the stopping size requirement for a given round.
+
+        Args:
+            sample_size (int): Current round size, i.e. number of ballots to be sampled in round
+
+        Returns:
+            An int which represents the minimum number of votes cast for the reported winner in the
+            current round size in order to stop the audit during that round.
+        """
+
+        if sample_size < 1:
+            raise ValueError('Sample size must be at least 1.')
+        if sample_size > self.contest.contest_ballots * self.max_fraction_to_draw:
+            raise ValueError(
+                'Sample size cannot be larger than max fraction to draw of the contest ballots.'
+            )
+
+        left = math.floor(sample_size / 2)
+        right = sample_size
+
+        while left < right:
+            proposed_stop = (left + right) // 2
+            proposed_stop_risk = self.compute_risk(proposed_stop, sample_size)
+
+            if proposed_stop_risk == self.alpha:
+                return proposed_stop
+
+            if proposed_stop_risk < self.alpha:
+                previous_stop_risk = self.compute_risk(proposed_stop - 1,
+                                                       sample_size)
+                if previous_stop_risk == self.alpha:
+                    return proposed_stop - 1
+                if previous_stop_risk > self.alpha:
+                    return proposed_stop
+                right = proposed_stop - 1
+            else:
+                left = proposed_stop + 1
+
+        # FIXME: Should we test final size when left = right and return some default if
+        # round size is too small?
+        return left
+
+    def compute_prior(self) -> np.ndarray:
         """Compute prior distribution of worst case election."""
 
-        left = np.zeros(self.contest.total_ballots_cast // 2, dtype=float)
+        half_contest_ballots = math.floor(self.contest.contest_ballots / 2)
+        left = np.zeros(half_contest_ballots, dtype=float)
         mid = np.array([0.5])
-        right = np.array(
-            [(0.5 / float(self.contest.total_ballots_cast // 2))
-             for i in range(self.contest.total_ballots_cast // 2)],
-            dtype=float)
+        right = np.array([
+            (0.5 / float(half_contest_ballots))
+            for i in range(self.contest.contest_ballots - half_contest_ballots)
+        ],
+                         dtype=float)
         return np.concatenate((left, mid, right))
 
     def compute_risk(self,
-                     sample: int = None,
+                     votes_for_winner: int = None,
                      current_round: int = None,
                      *args,
-                     **kwargs):
+                     **kwargs) -> float:
         """Compute the risk level given current round size and votes for winner in sample
 
-        TODO: insert description
+        The risk level is computed using the normalized product of the prior and posterior
```

Comment:
The wording here maybe should be adjusted; the posterior is the normalized product of the prior and the evidence / likelihood function.